src/tables.jl

```diff
@@ -213,11 +228,26 @@ function write(io_or_path, table, sv::SchemaVersion; validate::Bool=true,
         end
     end
     schema_metadata = LEGOLAS_SCHEMA_QUALIFIED_METADATA_KEY => identifier(sv)
+    provider_name, provider_version = schema_provider(sv)
+    provider_name_metadata = LEGOLAS_SCHEMA_PROVIDER_NAME_METADATA_KEY => string(provider_name)
+    provider_version_metadata = LEGOLAS_SCHEMA_PROVIDER_VERSION_METADATA_KEY => string(provider_version)
     if isnothing(metadata)
         metadata = (schema_metadata,)
+        if !isnothing(provider_name)
+            metadata = (metadata..., provider_name_metadata)
+            if !isnothing(provider_version)
+                metadata = (metadata..., provider_version_metadata)
+            end
+        end
```

Comment:
Should we add this to `schema_metadata` instead?

Reply:
I’m not sure tuples are supported if that’s what you mean